astropy/modeling/tests/test_polynomial.py

```diff
@@ -352,7 +351,7 @@ def test_polynomial_init_with_constraints(model_class):
 def test_sip_hst():
     """Test SIP against astropy.wcs"""
 
-    test_file = get_pkg_data_filename(os.path.join("data", "hst_sip.hdr"))
+    test_file = get_pkg_data_filename("data/hst_sip.hdr")
```

Comment:
I don't think the forward slash is correct on all systems. Can the function not accept a Path argument?

Reply:
The `get_pkg_data_*` do not work with path objects, I plan on creating a new PR to fix that. This PR I want to narrow down on making ruff happy.

> I don't think the forward slash is correct on all systems.

Yes but shouldn't the CI tests catch them? The windows CI test seem to be passing?

Reply:
> The get_pkg_data_* do not work with path objects, I plan on creating a new PR to fix that. This PR I want to narrow down on making ruff happy.

SGTM.

> Yes but shouldn't the CI tests catch them? The windows CI test seem to be passing?

Depends on exact usage. See https://docs.python.org/3/library/os.path.html#os.path.join for details.